mypy/nodes.py

```diff
@@ -2062,8 +2058,7 @@ def __init__(self, names: 'SymbolTable', defn: ClassDef, module_name: str) -> No
         self.assuming = []
         self.assuming_proper = []
         self.inferring = []
-        self._cache = set()
-        self._cache_proper = set()
+        TypeState.add_caches(self)
```

Comment:
What if this call would be optional? The TypeState API would be a bit simpler if `TypeState` would deal with adding entries as needed. Generally I'd prefer to keep public APIs as simple as possible and hide all logic within the implementation. Now the API is leaking an implementation detail (cache entry needs to exist for a class before it can be used for caches).